code-challenges.test.js

```diff
@@ -15,43 +15,144 @@
+//     expect(endTally(votes2)).toEqual("-31")
+//   })
+// })
+
+// const votes1 = { upVotes: 13, downVotes: 2 }
 // Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
+// const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
 
-// b) Create the function that makes the test pass.
+// FAIL  ./code-challenges.test.js
+// endTally
+//   ✕ takes in an object that contains up votes and down votes and returns the end tally (1 ms)
+// ● endTally › takes in an object that contains up votes and down votes and returns the end tally
+// ReferenceError: endTally is not defined
 
 // Pseudo code:
+// input: object, 2 key:value pairs
+// output: end tally
+// process: subtract the value of the keys
+        // dot notation can call on value
+        // can it subtract though?
+        // going deeper into classes.. maybe use a constructor
+        // am I overthinking this?
+
+// b) Create the function that makes the test pass.
+
+// const votes1 = { upVotes: 13, downVotes: 2 }
+// const votes2 = { upVotes: 2, downVotes: 33 }
+
+// const endTally = () => {
+
+// }
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
+// describe("noDupes", () => {
+//   it("returns one array with no duplicate values", () => {
+//     const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
+//     const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
+//     expect(noDupes(dataTypesArray1)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+//     expect(noDupes(dataTypesArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+//   })
+// })
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
-// b) Create the function that makes the test pass.
+// FAIL  ./code-challenges.test.js
+// noDupes
+//   ✕ returns one array with no duplicate values (1 ms)
+// ● noDupes › returns one array with no duplicate values
+// ReferenceError: noDupes is not defined
 
 // Pseudo code:
+// input: 2 arrays
+// output: 1 array
+// process: iteration to go through arrays
+// .filter() will filter through and return a subset, 
+// logical operator 
+// concat would join but stretch wants spread ...
+
+// const noDupes = (arr1, arr2) => {
+  // when using the .concat() I recieve and undefined w/the console
+  //  let newArr = arr1.concat(arr2)
+   // when attempting the spread I receive newArr is not defined
+  // let combined = [...arr1, ...arr2] 
+//   let output = []
+//   output = newArr.filter((value) =>{
+//   return output 
```

Comment:
This is a good approach and your thought process is on the right track. When you are writing out functions, you can throw in console.logs in the function, like we did during last assessment review, and it will log the data in the console when you run yarn jest. You can use this to check what you are getting from each step of the code. To troubleshoot further, I suggest bringing this up to a mentor so they can actually show you.